staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/webhook_config_interface.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiextensions
+
+import (
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/apiserver/pkg/util/webhook"
+)
+
+var _ webhook.ClientConfigInterface = &WebhookClientConfig{}
+
+func (c *WebhookClientConfig) GetURL() *string {
```

Comment:
Do we need this interface based on versioned or internal types? 

As written above, please move this where the interface implementation is actually used. This is not API.

Reply:
We cannot define an interface for a type outside of its package, I find it cleaner to implement it here though I can have a wrapper type and implement interface for that. Which one you think is better?

Reply:
Just use embedding: `type fooInterfaceImplementation struct { webhookClientConfig apiextensionsv1beta1 }`. Works fine in Go.